File: software/examples/python/example_callback.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

HOST = "localhost"
PORT = 4223
UID = "XYZ" # Change to your UID

from tinkerforge.ip_connection import IPConnection
from tinkerforge.bricklet_co2 import CO2

# Callback function for CO2 concentration callback (parameter has unit ppm)
def cb_co2_concentration(co2_concentration):
    print('CO2 Concentration: ' + str(co2_concentration) + ' ppm')

if __name__ == "__main__":
    ipcon = IPConnection() # Create IP connection
    co2 = CO2(UID, ipcon) # Create device object

    ipcon.connect(HOST, PORT) # Connect to brickd
    # Don't use device before ipcon is connected

    # Set period for CO2 concentration callback to 1s (1000ms)
    # Note: The CO2 concentration callback is only called every second
    #       if the CO2 concentration has changed since the last call!
    co2.set_co2_concentration_callback_period(1000)

    # Register CO2 concentration callback to function cb_co2_concentration
    co2.register_callback(co2.CALLBACK_CO2_CONCENTRATION, cb_co2_concentration)

    raw_input('Press key to exit\n') # Use input() in Python 3
    ipcon.disconnect()

File: software/examples/perl/example_callback.pl
#!/usr/bin/perl

use Tinkerforge::IPConnection;
use Tinkerforge::BrickletCO2;

use constant HOST => 'localhost';
use constant PORT => 4223;
use constant UID => 'XYZ'; # Change XYZ to the UID of your CO2 Bricklet

# Callback subroutine for CO2 concentration callback (parameter has unit ppm)
sub cb_co2_concentration
{
    my ($co2_concentration) = @_;

    print "CO2 Concentration: $co2_concentration ppm\n";
}

my $ipcon = Tinkerforge::IPConnection->new(); # Create IP connection
my $co2 = Tinkerforge::BrickletCO2->new(&UID, $ipcon); # Create device object

$ipcon->connect(&HOST, &PORT); # Connect to brickd
# Don't use device before ipcon is connected

# Register CO2 concentration callback to subroutine cb_co2_concentration
$co2->register_callback($co2->CALLBACK_CO2_CONCENTRATION, 'cb_co2_concentration');

# Set period for CO2 concentration callback to 1s (1000ms)
# Note: The CO2 concentration callback is only called every second
#       if the CO2 concentration has changed since the last call!
$co2->set_co2_concentration_callback_period(1000);

print "Press key to exit\n";
<STDIN>;
$ipcon->disconnect();
